api4/team.go

```diff
@@ -215,6 +216,34 @@ func updateTeamMemberRoles(c *Context, w http.ResponseWriter, r *http.Request) {
 	ReturnStatusOK(w)
 }
 
+func updateTeam(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireTeamId()
+	if c.Err != nil {
+		return
+	}
+	
+	team := model.TeamFromJson(r.Body)
+
+	if team == nil {
+		c.SetInvalidParam("team")
+		return
+	}
```

Comment:
Done. In order for me to test it, I added another param to the driver function. The teamId param is used for creating the right route, because if the id of the team was changed the wrong route was being accessed thus receiving the 'not authorized' error